packages/core/src/account/__tests__/simple.test.ts

```diff
@@ -102,29 +102,49 @@ describe("Account Simple Tests", () => {
       ]"
     `);
   });
-});
 
-const givenConnectedProvider = ({
-  owner,
-  chain,
-}: {
-  owner: SmartAccountSigner;
-  chain: Chain;
-}) =>
-  new SmartAccountProvider({
-    rpcProvider: `${chain.rpcUrls.alchemy.http[0]}/${"test"}`,
-    chain,
-  }).connect((provider) => {
+  it("should correctly use the account init code override", async () => {
     const account = new SimpleSmartContractAccount({
-      chain,
-      owner,
-      factoryAddress: getDefaultSimpleAccountFactoryAddress(chain),
-      rpcClient: provider,
+      chain: sepolia,
+      owner: owner,
+      factoryAddress: getDefaultSimpleAccountFactoryAddress(sepolia),
+      rpcClient: `${sepolia.rpcUrls.alchemy.http[0]}/${"test"}`,
+      // override the account address here so we don't have to resolve the address from the entrypoint
+      accountAddress: "0x1234567890123456789012345678901234567890",
```

Comment:
does this matter if this test is just evaluating initcode?

Reply:
yea because `getInitCode` calls `getAddress` which uses the `_getAccountInitCode` to call the entrypoint to check the revert for the address. this short circuits that